### PR TITLE
feat: cli support display more meaningful fields while list

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**
Before cli only show name, create time, id while list

**Solution:**
Support more fields while list/get

**Related Issue:**
#1170 
#1582

### CODE REVIEW

File: pkg/dao/types/resource.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ type (
 		// Name indicates the name of endpoint.
 		Name string `json:"name"`
 		// URL indicates the URL of endpoint, must be a valid URL.
-		URL string `json:"url"`
+		URL string `json:"url,cli-table-column"`
```
Comment (thxCode) [inline pkg/dao/types/resource.go:18]: we should move this change to the first commit of this PR.

**Comment by gitlawr** [comment]
How does `resource list` look like now?

**Comment by aiwantaozi** [comment]
> How does `resource list` look like now?

image

**Comment by gitlawr** [comment]
Can we show type(like the UI) and endpoints for resources?

File: pkg/apis/resource/basic_view.go [hunk 2]
```diff
@@ -662,3 +669,36 @@ func validateAttributesWithResourceDefinition(attrs property.Values, rd *model.R
...
+func genDisplayType(r *model.Resource) string {
+	if r.Type != "" {
+		return r.Type
+	}
+
+	switch {
+	case r.Edges.Template != nil && !r.Edges.Template.ProjectID.Valid():
+		return fmt.Sprintf("%s@%s (Global)", r.Edges.Template.Name, r.Edges.Template.Version)
+	default:
+		return fmt.Sprintf("%s@%s", r.Edges.Template.Name, r.Edges.Template.Version)
```
Comment (shanewxy) [inline pkg/apis/resource/basic_view.go:702]: Will panic if `r.Edges.Template == nil`?

File: pkg/apis/resource/basic_view.go [hunk 3]
```diff
@@ -32,14 +32,21 @@ import (
 	"github.com/seal-io/walrus/utils/validation"
 )
 
+type (
+	resourceOutput struct {
+		model.ResourceOutput `json:",inline"`
+		DisplayType          string `json:"displayType,omitempty,cli-table-column"`
```
Comment (gitlawr) [inline pkg/apis/resource/basic_view.go:38]: Can we keep API unchanged and do it on client side?

File: pkg/cli/formatter/table.go [hunk 4]
```diff
@@ -1,25 +1,51 @@
...
+	fieldStatus,
+	fieldCreateTime,
+}
+
+var fieldAlias = map[string]string{
+	fieldStatus:      "status",
+	fieldCreateTime:  "created",
+	fieldProject:     "project",
+	fieldEnvironment: "environment",
+	fieldDisplayType: "type",
```
Comment (gitlawr) [inline pkg/cli/formatter/table.go:42]: Capitalize table headers

**Review by gitlawr — APPROVED** [review]
LGTM

File: pkg/cli/formatter/table.go [hunk 5]
```diff
@@ -50,242 +76,226 @@ func (f *TableFormatter) Format(resp *http.Response) ([]byte, error) {
...
-		if !ok {
-			return nil, errors.New("can't decode response in table, use json or yaml format instead")
-		}
+	return []byte(formatted), nil
+}
 
-		if len(m) == 0 {
-			return []byte{}, nil
-		}
+func (f *TableFormatter) resourceItems(body []byte) string {
```
Comment (alexcodelf) [inline pkg/cli/formatter/table.go:100]: It seems that both `resourceItems` and `resourceItem` have similar logic to get table and it's string, shall them exact to a common function?

Comment (aiwantaozi) [reply pkg/cli/formatter/table.go:100]: Abstracted the common parts into function renderColumns with input columns and data.

**Review by alexcodelf — APPROVED** [review]
LGTM

**Review by alexcodelf — APPROVED** [review]

**Pull request merged**